borsh/src/de/mod.rs

```diff
@@ -514,6 +599,20 @@ where
     #[inline]
     fn deserialize_reader<R: Read>(reader: &mut R) -> Result<Self> {
         let vec = <Vec<T>>::deserialize_reader(reader)?;
+
+        #[cfg(feature = "de_strict_order")]
+        for pair in vec.windows(2) {
+            let [a, b] = pair else {
+                unreachable!("`windows` always return a slice of length 2 or nothing");
+            };
```

Comment:
Nice use of `let else` syntax ❤️ 

We may leave a todo to use `array_windows` once it stabilizes